src/nl/hannahsten/texifyidea/action/InsertEditorAction.kt

```diff
@@ -52,7 +52,9 @@ open class InsertEditorAction(
         // Don't touch any file content that is not related to TeXiFy
         if (file.psiFile(project)?.findElementAt(start)?.isLatexOrBibtex() == false) return
 
-        runWriteAction(project) { insert(document, start, end, editor.caretModel) }
+        if (file.isWritable) {
+            runWriteAction(project) { insert(document, start, end, editor.caretModel) }
```

Comment:
Why not put the `isWritable` check inside the `runWriteAction` function

Reply:
Because the `runWriteAction` did not have a reference to the file. I've added this as we only use it in three places, we should probably do the writingby altering the psi tree instead of writing to the document anyway.